bst.rb

```diff
@@ -0,0 +1,322 @@
+class String
+  def is_i?
+    /\A[-+]?\d+\z/ === self
+  end
+end
+
+class TreeNode
+  attr_accessor :value, :left, :right
+
+  def initialize(value)
+    @value = value
+    @left = nil
+    @right = nil
+  end
```

Comment:
you can add left and right as parameters in the function as it will help when you have to insert nodes in between

Reply:
added the variables to the parameters